### ISSUE TITLE
Can not install both Cartopy and GMT at the same time (conda)

### ISSUE THREAD

**Issue by MarkWieczorek** [SciTools/cartopy#1821]
### Description
Cartopy and GMT are both popular packages for map projections, so it would be good to ensure that these can cohabitate together. Some python packages in fact install both cartopy and gmt, so this is a major nuisance when the two have incompatible dependencies. 

When using conda, I have run into several problems in the past, and this appears to be related to GDAL. The most recent version of GMT (6.2.0) apepars to require '>=3.3.0,<3.4.0a0', whereas Cartopy (0.19) requires '>=3.1.3,<3.2.0a0'. I am confused as to where the Cartopy gdal dependency comes from, because it is not being installed explicitly by Cartopy (nor could I find it in the cartopy depencies either).

Given that many people use both of these packages, the developers of both teams should at least attempt to ensure that one package doesn't break the other. I am in contact with the GMT people and will see what can be done on their end (https://forum.generic-mapping-tools.org/t/can-not-install-both-gmt-and-cartopy-because-of-conda-dependency-conflicts/963). 

#### Code to reproduce
Here is the output when attempting to install gmt 6.2.0 and cartopy using conda with python 3.9. The conflict with gdal is explicit.
```
$ conda create -n test python=3.9 cartopy gmt=6.2.0

Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: -
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package libgdal conflicts for:
gmt=6.2.0 -> gdal[version='>=3.3.0,<3.4.0a0'] -> libgdal[version='3.3.0|3.3.0|3.3.0|3.3.0|3.3.1',build='hd181ca1_1|hd85aa46_2|hd51e85c_3|hd51e85c_1|hd51e85c_0|h9a52621_0']
cartopy -> gdal[version='>=3.1.3,<3.2.0a0'] -> libgdal[version='3.1.2|3.1.2|3.1.2|3.1.2|3.1.2|3.1.3|3.1.3|3.1.4',build='h6ef84d7_0|hd7bf8dc_1|h1f7cb0a_3|h55c8954_6|h0a7a322_9|hc84532a_12|h27e2542_15|hd51e85c_17|hd85aa46_16|hd181ca1_14|h9a52621_13|he0423de_11|h6de91e5_10|h62a8350_8|hbaa4b3a_7|hbdbaf39_5|he1d4179_4|hbb49e5d_2|h7b1ae08_1|hd7bf8dc_0|hd7bf8dc_0|hd7bf8dc_4|h6dfbaa8_3|hea8881a_2|h6dfbaa8_1']

Package python conflicts for:
python=3.9
gmt=6.2.0 -> gdal[version='>=3.3.0,<3.4.0a0'] -> python[version='>=3.6,<3.7.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0']

Package sqlite conflicts for:
cartopy -> proj[version='>=7.2.0,<7.2.1.0a0'] -> sqlite[version='3.13.*|3.20.*|>=3.24.0,<4.0a0|>=3.25.2,<4.0a0|>=3.25.3,<4.0a0|>=3.26.0,<4.0a0|>=3.28.0,<4.0a0|>=3.30.1,<4.0a0|>=3.32.3,<4.0a0|>=3.33.0,<4.0a0|>=3.34.0,<4.0a0|>=3.31.1,<4.0a0|>=3.29.0,<4.0a0|>=3.23.1,<4.0a0|>=3.22.0,<4.0a0|>=3.20.1,<4.0a0|>=3.25.1,<4.0a0|>=3.27.2,<4.0a0|>=3.35.4,<4.0a0|>=3.36.0,<4.0a0|>=3.35.5,<4.0a0|>=3.30.0,<4.0a0|3.9.*']
python=3.9 -> sqlite[version='>=3.33.0,<4.0a0|>=3.34.0,<4.0a0|>=3.35.5,<4.0a0|>=3.36.0,<4.0a0|>=3.35.4,<4.0a0']

Package numpy conflicts for:
cartopy -> numpy[version='1.10.*|1.11.*|1.12.*|1.13.*|>=1.11|>=1.14.6,<2.0a0|>=1.16.5,<2.0a0|>=1.16.6,<2.0a0|>=1.17.5,<2.0a0|>=1.18.5,<2.0a0|>=1.19.5,<2.0a0|>=1.19.4,<2.0a0|>=1.19.2,<2.0a0|>=1.9.3,<2.0a0|>=1.8|1.9.*|>=1.11.3,<2.0a0']
cartopy -> matplotlib-base[version='>=1.5.1'] -> numpy[version='1.8.*|>=1.15.1,<2.0a0|>=1.15.4,<2.0a0|>=1.18.1,<2.0a0|>=1.9|>=1.7|>=1.14.1,<2.0a0']
```

And here is the output when using python 3.8. Here, it is not obvious what the problem is
```
conda create -n test python=3.8 cartopy gmt=6.2.0

Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: |
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package numpy conflicts for:
cartopy -> numpy[version='1.10.*|1.11.*|1.12.*|1.13.*|>=1.11|>=1.14.6,<2.0a0|>=1.16.5,<2.0a0|>=1.16.6,<2.0a0|>=1.17.5,<2.0a0|>=1.18.5,<2.0a0|>=1.19.5,<2.0a0|>=1.19.4,<2.0a0|>=1.19.2,<2.0a0|>=1.9.3,<2.0a0|>=1.8|1.9.*|>=1.11.3,<2.0a0']
cartopy -> matplotlib-base[version='>=1.5.1'] -> numpy[version='1.8.*|>=1.15.1,<2.0a0|>=1.15.4,<2.0a0|>=1.18.1,<2.0a0|>=1.9|>=1.7|>=1.14.1,<2.0a0']

Package libpng conflicts for:
cartopy -> matplotlib-base[version='>=1.5.1'] -> libpng[version='>=1.6.23,<1.7|>=1.6.35,<1.7.0a0|>=1.6.37,<1.7.0a0|>=1.6.36,<1.7.0a0|>=1.6.34,<1.7.0a0|>=1.6.32,<1.7.0a0']
gmt=6.2.0 -> graphicsmagick -> libpng[version='>=1.6.34,<1.7.0a0|>=1.6.35,<1.7.0a0|>=1.6.37,<1.7.0a0']

Package libcxxabi conflicts for:
python=3.8 -> libcxx[version='>=4.0.1'] -> libcxxabi[version='4.0.1|4.0.1|8.0.0|8.0.0|8.0.0|8.0.0|8.0.1',build='hebd6815_0|3|4|0|2|1|hcfea43d_1']
cartopy -> libcxx[version='>=4.0.1'] -> libcxxabi[version='4.0.1|4.0.1|8.0.0|8.0.0|8.0.0|8.0.0|8.0.1',build='hebd6815_0|3|4|0|2|1|hcfea43d_1']

Package python conflicts for:
gmt=6.2.0 -> gdal[version='>=3.3.0,<3.4.0a0'] -> python[version='>=3.6,<3.7.0a0|>=3.9,<3.10.0a0|>=3.7,<3.8.0a0|>=3.8,<3.9.0a0']
python=3.8
```
#### Traceback 

```

```

<details>
 <summary>Full environment definition</summary>

### conda list

```
```

### pip list

```
```

</details>

**Comment by akrherz** [comment]
Cartopy is going to "jam up" many conda environments due to its older `proj` requirement.  The excellent and hard work with #1808 will hopefully break this log jam once merged and with the next cartopy release.

**Comment by MarkWieczorek** [comment]
Ok. Thanks. But do you know what the problem is? Proj doesn't appear to have gdal as a dependency.

**Comment by akrherz** [comment]
You probably want to engage the [conda-forge/cartopy](https://github.com/conda-forge/cartopy-feedstock) repo then and see if the gdal pin can be relaxed / moved forward or what may be going on.

**Comment by xylar** [comment]
The lack of support of `proj` 8 is unfortunately making it increasingly hard to include `cartopy` in any functional stack of scientific python packages on conda-forge.  Nearly all new builds of other packages are with `proj` 8, meaning they cannot be installed with `cartopy`.

**Comment by greglucas** [comment]
@xylar, are you able to create a test conda-forge label release with the latest master branch to test out the recent update? We have now merged the pyproj effort, which allows builds with proj > 8.

**Comment by xylar** [comment]
@greglucas, that's very exciting!  I would be glad to make a test build as soon as I'm able.  I'll try to do that tomorrow but it might have to wait a little while.

**Comment by xylar** [comment]
I'm working on this here, in case anyone wants to follow along:
https://github.com/conda-forge/cartopy-feedstock/pull/117

The only "issue" I've come across so far is deciding what to call the build.  For now, it's 0.20.0a1 but I'm open to suggestions.  It would be nice to put it on the `cartopy_dev` so other packages can be tested with it (using `-c conda-forge/label/cartopy_dev cartopy=0.20.0a1`)

**Comment by xylar** [comment]
I think I've finally got it worked out.  We'll need to drop python<3.7 because there simply aren't compatible `proj` and `pyproj` builds available for python 3.6.  Other than that, we seem to be in good shape.  It would also be good to test with proj 8.1.0 and PyPy for Windows, since those PRs aren't working with the current release.

**Comment by QuLogic** [comment]
Cartopy now supports proj >= 8 (and only that) with 0.20, and it appears to already be in conda-forge. As this is an external packaging issue, there's nothing more to be done here.
